### PR TITLE
Adding optional sampler input to DataModule

### PR DESCRIPTION
**Pull request by lillekemiker**
## What does this PR do?

This PR adds the option of specifying a sampler to be used in the training `DataLoader` returned by a `DataModule`.

Fixes # [350](https://github.com/PyTorchLightning/lightning-flash/issues/350)

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by lillekemiker** [comment]
Could someone supply me with info on how I should add this to the `CHANGELOG.md`? Considering that it technically changes the user facing API, I guess the semantic version should be bumped on the "minor" level, but that just seems big for this pretty small change.

**Review by ethanwharris — APPROVED** [review]
@lillekemiker Awesome! LGTM. Regarding the CHANGELOG, could you just add a line under `Added` in the `Unreleased` section at the top, no need for a version bump :smiley:

**Review by Borda — APPROVED** [review]

File: flash/core/data/data_module.py [hunk 1]
```diff
@@ -259,11 +264,21 @@ def _resolve_collate_fn(self, dataset: Dataset, running_stage: RunningStage) ->
...
+            shuffle = not isinstance(train_ds, (IterableDataset, IterableAutoDataset))
+            return DataLoader(
+                train_ds,
+                batch_size=self.batch_size,
+                shuffle=shuffle,
+                num_workers=self.num_workers,
+                pin_memory=True,
+                drop_last=True,
+                collate_fn=self._resolve_collate_fn(train_ds, RunningStage.TRAINING)
+            )
```
Comment (edgarriba) [inline flash/core/data/data_module.py:277]: I'd simply similar to this
```python
shuffle: bool = False
if self.sampler is None:
  shuffle = not isinstance(train_ds, (IterableDataset, IterableAutoDataset))
return DataLoader(
   train_ds,
   batch_size=self.batch_size,
   shuffle=shuffle,
   sampler=self.sampler,
   num_workers=self.num_workers,
   pin_memory=True,
   drop_last=True,
   collate_fn=self._resolve_collate_fn(train_ds, RunningStage.TRAINING)
)
```

**Review by edgarriba — APPROVED** [review]
LGTM. Just small suggestion

File: flash/core/data/data_module.py (hunk 1 above)
Comment (lillekemiker) [reply flash/core/data/data_module.py]: Done!

**Pull request merged**